pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -193,6 +196,8 @@ private PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopG
                     new ExecutorProvider(conf.getNumListenerThreads(), "pulsar-external-listener");
             this.internalExecutorProvider = internalExecutorProvider != null ? internalExecutorProvider :
                     new ExecutorProvider(conf.getNumIoThreads(), "pulsar-client-internal");
+            this.scheduledExecutorProvider = new ScheduledExecutorProvider(conf.getNumIoThreads(),
```

Comment:
@codelipenghui - I think we might want to provide a way for end users (and brokers) to supply this `ScheduledExecutorProvider`. We do that in the broker for the `externalExecutorProvider` and the `internalExecutorProvider` on the lines above to limit the resources required for multiple clients that run within the broker.

Reply:
Thank you @codelipenghui.